execution/src/execution.rs

```diff
@@ -345,6 +346,102 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
         }
         Ok(logs)
     }
+
+    pub async fn get_fee_history(
+        &self,
+        block_count: u64,
+        last_block: u64,
+        reward_percentiles: &[f64],
+        payloads: &BTreeMap<u64, ExecutionPayload>,
+    ) -> Result<Option<FeeHistory>> {
+        let helios_latest_block = *payloads.last_key_value().unwrap().0;
+        let helios_oldest_block = *payloads.first_key_value().unwrap().0;
+        let mut block_count = block_count;
+        let mut request_latest_block = last_block;
+
+        debug!(
+            "get_fee_history parameters:
+               helios_latest_block {:?}, helios_oldest_block {:?},
+               block_count {:?}, request_latest_block {:?}, reward_percentiles {:?}",
+            helios_latest_block,
+            helios_oldest_block,
+            block_count,
+            request_latest_block,
+            reward_percentiles
+        );
+
+        //case: requested block is more recent than helios' newest block
+        if request_latest_block > helios_latest_block {
+            request_latest_block = helios_latest_block;
+        }
+
+        //Can't prove anything in the request range
+        if request_latest_block < helios_oldest_block {
+            return Ok(None);
```

Comment:
I would return an error here personally - the end consumer should know if their request is out of the block range.